daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -240,6 +240,27 @@ convertInt64 x
     | otherwise =
         unsupported "Int literal out of bounds" (negate x)
 
+convertRational :: Integer -> Integer -> ConvertM LF.Expr
+convertRational num denom
+ =
+    -- the denominator needs to be a divisor of 10^10.
+    -- num % denom * 10^10 needs to fit within a 128bit signed number.
```

Comment:
The spec says we need to check that
```
abs (num / denom) < 10^28
```
When `10^10 mod denom == 0`, this translates to
```
abs x < 10^38
```
for
```
x = num * 10^10 div denom
```
When this condition is met, the `Decimal` we want is `MkFixed x :: Fixed E10`.